magento/magento2: #8765

cavalier79 opened the pull request:
M2.1.4 : Magento\Catalog\Model\ResourceModel\AbstractResource::getAttributeRawValue() returns false negative

- fixed adding a strong check validation on return value.

Commit:
magento/magento2 #2370: A PHPDoc comment for \Magento\Config\Model\Config\SourceFactory::create() method is wrong

Fixed type in return comment

Commit:
magento/magento2 #2370:

A PHPDoc comment for \Magento\Config\Model\Config\SourceFactory::create() method is wrong
The correct return parameter is  \Magento\Framework\Option\ArrayInterface

Commit:
magento/magento2 #8765:

Magento\Catalog\Model\ResourceModel\AbstractResource::getAttributeRawValue() returns false negative
- fixed adding a strong check on return value

orlangur commented on app/code/Magento/Catalog/Model/ResourceModel/AbstractResource.php:
```diff
@@ -568,6 +568,6 @@ public function getAttributeRawValue($entityId, $attribute, $store)
         }
 
-        return $attributesData ? $attributesData : false;
+        return $attributesData!==false ? $attributesData : false;
```
Please add spaces around binary operator.

Also, as to me this statement would be less confusing if you invert condition (`=== false` and swap ternary operator parts).

Commit:
magento/magento2 #8765:

Magento\Catalog\Model\ResourceModel\AbstractResource::getAttributeRawValue() returns false negative
- fixed adding a strong check on return value

orlangur commented on app/code/Magento/Catalog/Model/ResourceModel/AbstractResource.php:
```diff
@@ -568,6 +568,6 @@ public function getAttributeRawValue($entityId, $attribute, $store)
         }
 
-        return $attributesData ? $attributesData : false;
+        return $attributesData === false ? false: $attributesData;
```
Thanks! Just one more space in `false:` left to add :)

It would be really nice if you squash all changes into one commit and force push into the same branch, but I do not insist on this.

cavalier79 replied:
Fixed.. but i've still have some problem with squash :D

Commit:
magento/magento2 #8765:

Magento\Catalog\Model\ResourceModel\AbstractResource::getAttributeRawValue() returns false negative
- fixed adding a strong check on return value

vrann commented:
@cavalier79 Thank you, accepted.
